ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt

```diff
@@ -85,30 +77,31 @@ private fun Template2MainContent(state: PaywallViewState.Template2, viewModel: P
         horizontalAlignment = Alignment.CenterHorizontally,
         verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
     ) {
-        IconImage(paywallData = state.paywallData)
-        val localizedConfig = state.paywallData.localizedConfig()
+        state.templateConfiguration.images.iconUri?.let { IconImage(it) }
+        val localizedConfig = state.templateConfiguration.packageConfiguration.defaultInfo.localization
+        val colors = state.templateConfiguration.getColors(isSystemInDarkTheme())
         Text(
             style = MaterialTheme.typography.displaySmall,
             fontWeight = FontWeight.Black,
             textAlign = TextAlign.Center,
-            text = localizedConfig?.title ?: "",
-            color = state.paywallData.getColors().text1Color,
+            text = localizedConfig.title,
+            color = colors.text1,
         )
         Text(
             style = MaterialTheme.typography.bodyLarge,
             fontWeight = FontWeight.Bold,
             textAlign = TextAlign.Center,
-            text = localizedConfig?.subtitle ?: "",
-            color = state.paywallData.getColors().text1Color,
+            text = localizedConfig.subtitle ?: "",
+            color = colors.text1,
         )
-        state.packages.forEach { aPackage ->
-            SelectPackageButton(state, aPackage, viewModel)
+        state.templateConfiguration.packageConfiguration.allPackagesInfo.forEach { packageInfo ->
```

Comment:
With my suggestion this would become a bit simpler
```suggestion
        state.templateConfiguration.packages.all.forEach { packageInfo ->
```